packages/core/src/Stores/contract-store.js

```diff
@@ -433,6 +484,15 @@ function calculate_marker(contract_info, { accu_high_barrier, accu_low_barrier }
     if (tick_count >= 1) {
```

Comment:
lets clean up this logic here

```
if(!tick_count) {
// getEndTime returns undefined when contract is running.
const end_time = getEndTime(contract_info) || date_expiry;
const epoch_array = [date_start, end_time];

if (entry_tick_time) epoch_array.push(entry_tick_time);
if (exit_tick_time) epoch_array.push(exit_tick_time);

const non_tick_contract_Info = {
  contract_info: toJS(contract_info),
  type: 'NonTickContract',
  key: `${contract_id}-date_start`,
  epoch_array,
  price_array,
};

if (isResetContract(contract_type)) {
  non_tick_contract_Info.epoch_array = [reset_time, ...epoch_array];
  non_tick_contract_Info.price_array = [price_array[0], reset_barrier];
}

return non_tick_contract_Info;
```

Reply:
```
if(tick_count >= 1) {
const commonData = {
    contract_info: toJS(contract_info),
    type: 'TickContract',
    key: `${contract_id}-date_start`,
    epoch_array: [date_start, ...ticks_epoch_array],
    price_array,
};

if (isDigitContract(contract_type)) {
    // DigitContract
    return {
        ...commonData,
        type: 'DigitContract',
    };
}

if (isResetContract(contract_type)) {
    return {
        ...commonData,
        epoch_array: [reset_time, ...ticks_epoch_array],
        price_array: [price_array[0], reset_barrier],
    };
}
}
return commonData;
```

Reply:
just double check and see if the logic is correct 👍

Reply:
Nice, I didn't thought about refactoring this part, thank for valuable point, will do

Reply:
In this case even `tick_count >= 1` can be simplify to `if (tick_count){ ... }`